mlir/lib/Dialect/XeGPU/IR/XeGPUOps.cpp

```diff
@@ -191,11 +191,21 @@ IsValidMatrixOpParams(VectorType dataTy, MemDescType mdescTy,
 
   ArrayRef<int64_t> dataShape = dataTy.getShape();
   ArrayRef<int64_t> mdescShape = mdescTy.getShape();
-
+  if (subgroup_block_io && layout) {
+    auto laneData = layout.getEffectiveLaneDataAsInt();
+    if (!laneData.empty()) {
+      bool isLaneDataLinear =
+          std::all_of(laneData.begin(), std::prev(laneData.end()),
+                      [](int x) { return x == 1; });
+      if (!isLaneDataLinear)
+        return emitError()
+               << "With subgroup_block_io, lane data must be linear.";
+      if (isLaneDataLinear && laneData.back() != 1)
```

Comment:
Not sure about this condition.
I think here it verify the condition that "For operation without subgroup_block_io, the lane_data must be physically contiguous in the slm."
1) First of all, why isLaneDataLinear needed here? 
2) It should check the !subgroup_block_io. 
3) It then check for the dimension n that laneData is not 1, stride[n] must be 1.